lib/mix/tasks/usage_rules.sync.ex

```diff
@@ -945,6 +970,201 @@ if Code.ensure_loaded?(Igniter) do
+                [{package_name, package_path, nil}]
+              else
+                []
+              end
+
+            sub_rules =
+              find_available_sub_rules(igniter, package_path)
+              |> Enum.map(fn sub_rule_name ->
+                {package_name, package_path, sub_rule_name}
+              end)
+
+            main_rules ++ sub_rules
+          end)
+        else
+          expanded_packages = expand_wildcard_specs(igniter, all_deps, provided_packages)
+
+          expanded_packages
+          |> Enum.flat_map(fn package_spec ->
+            {package_name, sub_rule} = parse_package_spec(package_spec)
+
+            case Enum.find(all_deps, fn {name, _path} -> name == package_name end) do
+              {_name, package_path} ->
+                case sub_rule do
+                  nil ->
+                    usage_rules_path = Path.join(package_path, "usage-rules.md")
+
+                    if Igniter.exists?(igniter, usage_rules_path) do
+                      [{package_name, package_path, nil}]
+                    else
+                      []
+                    end
+
+                  sub_rule_name ->
+                    sub_rule_path =
+                      Path.join([package_path, "usage-rules", "#{sub_rule_name}.md"])
+
+                    if Igniter.exists?(igniter, sub_rule_path) do
+                      [{package_name, package_path, sub_rule_name}]
+                    else
+                      []
+                    end
+                end
+
+              nil ->
+                []
+            end
+          end)
+        end
+
+      # Generate Claude Skills format for each package
+      Enum.reduce(packages_to_process, igniter, fn {name, path, sub_rule}, acc ->
+        usage_rules_path =
+          case sub_rule do
+            nil ->
+              Path.join(path, "usage-rules.md")
+
+            sub_rule_name ->
+              Path.join([path, "usage-rules", "#{sub_rule_name}.md"])
+          end
+
+        content =
+          case Rewrite.source(acc.rewrite, usage_rules_path) do
+            {:ok, source} -> Rewrite.Source.get(source, :content)
+            {:error, _} -> File.read!(usage_rules_path)
+          end
+
+        skill_name =
+          case sub_rule do
+            nil ->
+              name
+              |> to_string()
+              |> String.replace("_", "-")
+
+            sub_rule_name ->
+              base_name =
+                name
+                |> to_string()
+                |> String.replace("_", "-")
+
+              formatted_sub =
+                sub_rule_name
+                |> String.replace("_", "-")
+
+              "#{base_name}-#{formatted_sub}"
+          end
+
+        # Extract description from first heading (H1 or H2) or first line of text
+        # Ignore HTML comments when finding the first line
+        description =
+          content
+          |> String.split("\n")
```

Comment:
talked about it in discord, but for this to make sense I think it should be some text suited to this purpose, either the package description plus some static text, or configured by the library providing these rules.